### DISCUSSION TITLE
Don't skip validate by default

### DISCUSSION THREAD

**Discussion by mikicho** [testjavascript/nodejs-integration-tests-best-practices#154] [General]
I want to appeal [this decision](https://github.com/testjavascript/nodejs-integration-tests-best-practices/blob/master/example-application/test/basic-tests.test.js#L16).
Currently, we recommend overriding Axios' default behavior to not validate the response status code.

--------

I find this behavior inconvenient and error-prone because we're using the same Axios client [also for the `arrange` phase](https://github.com/testjavascript/nodejs-integration-tests-best-practices/blob/master/example-application/test/basic-tests.test.js#L58) which results to unexpected behavior of the test.
For example, [this](https://github.com/testjavascript/nodejs-integration-tests-best-practices/blob/dd97923e077745e17d99ae2f312dfacbf4efe90a/recipes/data-isolation/data-isolation.test.js#L117) test may pass even if one of the requests in the `arrange` phase fails:

```javascript
    test('When deleting an existing order, Then it should NOT be retrievable', async () => {
      // Arrange
      const orderToDelete = {
        userId: 1,
        productId: 2,
        externalIdentifier: `id-${getShortUnique()}`, 
      };
      const deletedOrder = (await axiosAPIClient.post('/order', orderToDelete)).data.id; // this returns 500, deletedOrder is undefined
      const orderNotToBeDeleted = orderToDelete;
      orderNotToBeDeleted.externalIdentifier = `id-${getShortUnique()}`;
      const notDeletedOrder = (await axiosAPIClient.post('/order', orderNotToBeDeleted)).data.id;

      // Act
      const deleteRequestResponse = await axiosAPIClient.delete(`/order/${deletedOrder}`); // This may returns 204, depends on the implementation
      

      // Assert
      const getDeletedOrderStatus = (await axiosAPIClient.get(`/order/${deletedOrder}`)).status;
      const getNotDeletedOrderStatus = (await axiosAPIClient.get(`/order/${notDeletedOrder}`)).status;
      expect(getNotDeletedOrderStatus).toBe(200);
      expect(getDeletedOrderStatus).toBe(404);
      expect(deleteRequestResponse.status).toBe(204);
    });
  });
```

Another point, this reduce test "debugability" in complex `arrange` scenarios because we don't "fail-fast" and you need to figure out the root cause:

```javascript
test('...', () => {
  // Arrange
  const a = await createA();
  const b = await createB(a); // returns 500
  const c = await createC(b);
  const d = await createD(c);
  
  // Act
  const response = await axiosAPIClient.delete(`/order/${d.id}`); // This returns 500
      

  // Assert
  expect(response.status).toBe(200);
})
```
 
This test result is `expected 200, got 500`, which hides the real root cause.

The solutions I see:
1. Don't override the Axios default and set it for each test:
```
const response = await axiosAPIClient.delete(`/order/${d.id}`, { validateStatus: () => true }); 
```
2. Create two Axios instances, one for `arrange` and one for `act`.

**Comment by rluvaton** [comment]
I think the first solution is good in the way that the test is a self-contained (having everything it needs in it) which I think we should aim for in most cases, but adds a lot of code to the test making it less readable

The 2nd solution although may raise eyebrows but I think it's the best solution as in the majority of the tests you want to have a successful request in the `arrange` part

So I would go with the 2nd option
